extensions/kubeapi/rbac/update.go

```diff
@@ -42,3 +42,71 @@ func UpdateGlobalRole(client *rancher.Client, updatedGlobalRole *v3.GlobalRole)
 	}
 	return newGlobalRole, nil
 }
+
+// UpdateRoleTemplate is a helper function that uses the dynamic client to update an existing cluster role template
+func UpdateRoleTemplate(client *rancher.Client, updatedCRT *v3.RoleTemplate) (*v3.RoleTemplate, error) {
+	dynamicClient, err := client.GetDownStreamClusterClient(LocalCluster)
+	if err != nil {
+		return nil, err
+	}
+	crtUnstructured := dynamicClient.Resource(RoleTemplateGroupVersionResource)
+	roleTemplate, err := crtUnstructured.Get(context.TODO(), updatedCRT.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	currentCRT := &v3.RoleTemplate{}
+	err = scheme.Scheme.Convert(roleTemplate, currentCRT, roleTemplate.GroupVersionKind())
+	if err != nil {
+		return nil, err
+	}
+
+	updatedCRT.ObjectMeta.ResourceVersion = currentCRT.ObjectMeta.ResourceVersion
+
+	unstructuredResp, err := crtUnstructured.Update(context.TODO(), unstructured.MustToUnstructured(updatedCRT), metav1.UpdateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	newCRT := &v3.RoleTemplate{}
+	err = scheme.Scheme.Convert(unstructuredResp, newCRT, unstructuredResp.GroupVersionKind())
+	if err != nil {
+		return nil, err
+	}
+	return newCRT, nil
+
+}
+
+// UpdateClusterRoleTemplateBindings is a helper function that uses the dynamic client to update an existing cluster role template binding
+func UpdateClusterRoleTemplateBindings(client *rancher.Client, namespace string, updatedCRTB *v3.ClusterRoleTemplateBinding) (*v3.ClusterRoleTemplateBinding, error) {
+	dynamicClient, err := client.GetDownStreamClusterClient(LocalCluster)
+	if err != nil {
+		return nil, err
+	}
+	crtbUnstructured := dynamicClient.Resource(ClusterRoleTemplateBindingGroupVersionResource).Namespace(namespace)
+	clusterRoleTemplateBinding, err := crtbUnstructured.Get(context.TODO(), updatedCRTB.Name, metav1.GetOptions{})
```

Comment:
Same comment as https://github.com/rancher/shepherd/pull/15#discussion_r1442098919 :)
Wouldn't this error out if one tries to update the crtb name as they will get no crtb found?